core/src/main/java/org/mule/runtime/core/source/polling/schedule/ScheduledPoll.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.source.polling.schedule;
+
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.lifecycle.InitialisationException;
+import org.mule.runtime.api.lifecycle.Lifecycle;
+import org.mule.runtime.api.meta.NameableObject;
+import org.mule.runtime.core.api.scheduler.Scheduler;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+
+/**
+ * Definition of a Scheduler for poll.
+ *
+ * @since 3.5.0, moved from {@link org.mule.runtime.core.api.schedule.Scheduler}.
+ */
+public class ScheduledPoll implements Lifecycle, NameableObject {
```

Comment:
Again with this name, any reason it shouldn't be called something broader than "poll"?  To poll is just one reason you might need to schedule a recurrent task.  Also, this should be with scheduler infrastructure rather than in package with polling source shouldn't it?